samples/drivers/gecko_adc/src/bq24250.c

```diff
@@ -0,0 +1,330 @@
+	if (ret) {
+		printf("i2c read_bytes function:Error reading from BQ! error code (%d)\n", ret);
+	}
+
+	ret = i2c_read_to_buffer(bq_addr, bq_reg_0x01, &bq_reg_0x01_data, rx_len);
+	if (ret) {
+		printf("i2c read_bytes function:Error reading from BQ! error code (%d)\n", ret);
+	}
+
+	k_msleep(100);
+
+	ret = i2c_read_to_buffer(bq_addr, bq_reg_0x00, &bq_reg_0x00_data, rx_len);
+	if (ret) {
+		printf("i2c read_bytes function:Error reading from BQ! error code (%d)\n", ret);
+	}
+
+	ret = i2c_read_to_buffer(bq_addr, bq_reg_0x01, &bq_reg_0x01_data, rx_len);
+	if (ret) {
+		printf("i2c read_bytes function:Error reading from BQ! error code (%d)\n", ret);
+	}
+
+	printf("\nReceived a Charge Status or Fault Interrupt callback\n");
+
+	/* parse the BQ Charge Status or Fault Interrupt that got us here. */
+	uint8_t charger_status = ((bq_reg_0x00_data & 0x30) >> 4);
+	uint8_t charge_fault_status = (bq_reg_0x00_data & 0x0F);
+
+	if ((charger_status == BQ_CHARGER_FAULT_MODE) &&
+		((charge_fault_status == BQ_SLEEP_MODE) ||
+		 (charge_fault_status == INPUT_FAULT_AND_LDO_LOW_MODE))) {
+		if (!no_vbus) {
+			printf("\tCharger VBUS has just been removed!\n");
+		} else {
+			printf("\tCharger VBUS is missing!\n");
+		}
+		no_vbus = true;
+	} else {
+		no_vbus = false;
+		switch (charger_status) {
+		case 0:
+			printf("\tCharger is ready to charge!\n");
+			break;
+		case 1:
+			printf("\tCharger is currently charging!\n");
+			break;
+		case 2:
+			printf("\tCharger is done charging!\n");
+			break;
+		case 3:
+			printf("\tCharger is in Fault mode!\n");
+			break;
+		default:
+			break;
+		}
+
+		switch (charge_fault_status) {
+		case 0:
+			printf("\tCharger is in normal mode!\n");
+			break;
+		case 1:
+			printf("\tCharger is in Input OVP mode!\n");
+			break;
+		case 2:
+			printf("\tCharger is in Input UVLO mode!\n");
+			break;
+		case 3:
+			printf("\tCharger is in Sleep mode!\n");
+			break;
+		case 4:
+			printf("\tBattery is in Temperature Fault mode!\n");
+			break;
+		case 5:
+			printf("\tBattery is in OVP mode!\n");
+			break;
+		case 6:
+			printf("\tBattery is in Thermal Shutdown mode!\n");
+			break;
+		case 7:
+			printf("\tBattery is in Timer Fault!\n");
+			break;
+		case 8:
+			printf("\tBattery is in No Battery Connected ");
+			printf("mode - Try reseating the connector!\n");
+			break;
+		case 9:
+			printf("\tBattery is in ISET short mode!\n");
+			break;
+		case 10:
+			printf("\tBattery is in Input Fault and LDO low mode!\n");
+			break;
+		default:
+			break;
+		}
+	}
+	return ret;
+}
+
+static void bq_intr_callback(const struct device *port,
+		struct gpio_callback *cb, uint32_t pins)
+{
+	bq_battery_charge_int_isr_count++;
+	bq_battery_charge_int_state_change = true;
+}
+
+static void bq_thread(void *a, void *b, void *c)
+{
+	ARG_UNUSED(a);
+	ARG_UNUSED(b);
+	ARG_UNUSED(c);
+
+	k_msleep(4000);
+
+	printf("%s - thread started\n", __func__);
+	uint8_t current_state = BQ_INIT;
+	int ret;
+
+	memset(i2c_read_data, 0, 20);
```

Comment:
The hardcoded value of 20 should be replaced with `sizeof(i2c_read_data)`.

Reply:
Done